Client/Frontend/Browser/BrowserViewController/BrowserViewController+Menu.swift

```diff
@@ -153,9 +153,28 @@ extension BrowserViewController {
                 }
             }
             MenuItemButton(icon: #imageLiteral(resourceName: "menu-settings").template, title: Strings.settingsMenuItem) { [unowned self, unowned menuController] in
-                let keyringStore = BraveWallet.KeyringControllerFactory
-                    .get(privateMode: PrivateBrowsingManager.shared.isPrivateBrowsing)
-                    .map { KeyringStore(keyringController: $0) }
+                let privateMode = PrivateBrowsingManager.shared.isPrivateBrowsing
+                guard
+                    let keyringController = BraveWallet.KeyringControllerFactory.get(privateMode: privateMode),
+                    let rpcController = BraveWallet.EthJsonRpcControllerFactory.get(privateMode: privateMode),
+                    let assetRatioController = BraveWallet.AssetRatioControllerFactory.get(privateMode: privateMode),
+                    let walletService = BraveWallet.ServiceFactory.get(privateMode: privateMode),
+                    let swapController = BraveWallet.SwapControllerFactory.get(privateMode: privateMode),
+                    let txController = BraveWallet.EthTxControllerFactory.get(privateMode: privateMode)
+                else {
+                    log.error("Failed to load wallet. One or more services were unavailable")
+                    return
+                }
```

Comment:
We don't want settings to fail to load if wallet settings aren't available, we need to make a WalletStore optionally created based on availability of the services and if it doesn't exist then settings just doesn't add a wallet row